src/functional/otf/compilation/build_systems/compiledb.py

```diff
@@ -0,0 +1,293 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+import dataclasses
+import json
+import pathlib
+import subprocess
+from typing import Optional
+
+from functional.otf import languages, stages
+from functional.otf.compilation import build_data, cache, compiler
+from functional.otf.compilation.build_systems import cmake, cmake_lists
+from functional.otf.source import source
+
+
+@dataclasses.dataclass
+class CompiledbFactory(
+    compiler.BuildSystemProjectGenerator[
+        languages.Cpp, languages.LanguageWithHeaderFilesSettings, languages.Python
+    ]
+):
+    """
+    Create a CompiledbProject from a ``CompilableSource`` stage object with given CMake settings.
+
+    Use CMake to generate a compiledb with the required sequence of build commands.
+    Generate a compiledb only if there isn't one for the given combination of cmake configuration and library dependencies.
+    """
+
+    cmake_build_type: str = "Debug"
+    cmake_extra_flags: Optional[list[str]] = None
+    renew_compiledb: bool = False
+
+    def __call__(
+        self,
+        source: stages.CompilableSource[
+            languages.Cpp,
+            languages.LanguageWithHeaderFilesSettings,
+            languages.Python,
+        ],
+        cache_strategy: cache.Strategy,
+    ) -> CompiledbProject:
+        if not source.binding_source:
+            raise compiler.CompilerError(
+                "CMake build system project requires separate bindings code file."
+            )
```

Comment:
If that's so, I'm happy to remove it.

Reply:
I had simply taken this over from the `CMakeFactory`, which does explicitly access the bindings and therefore needs this. So does this code. To avoid checking for existence of bindings and implementing code without a use case I will leave this as it is. Whenever there is a usecase it will be necessary to adapt this code and remove the check. At that time there will also be something to test it against.

Reply:
What is true is that the reason for the check here is that the `Compiledb` build system requires it. For now. Therefore the message should be adapted.

Reply:
Done